src/main/scala/com/databricks/labs/overwatch/pipeline/InitializerFunctions.scala

```diff
@@ -400,69 +400,26 @@ trait InitializerFunctions
-        }
-
-        if (auditLogPath.nonEmpty)
-          dbutils.fs.ls(auditLogPath.get).foreach(auditFolder => {
-            if (auditFolder.isDir) require(auditFolder.name.startsWith("date="), s"Audit directory must contain " +
-              s"partitioned date folders in the format of ${auditLogPath.get}/date=. Received ${auditFolder} instead.")
-          })
-
-        val supportedAuditLogFormats = Array("json", "parquet", "delta")
-        if (!supportedAuditLogFormats.contains(auditLogFormat)) {
-          throw new BadConfigException(s"Audit Log Format: Supported formats are ${supportedAuditLogFormats.mkString(",")} " +
-            s"but $auditLogFormat was placed in teh configuration. Please select a supported audit log format.")
-        }
-
-        val finalAuditLogPath = if (auditLogPath.get.endsWith("/")) auditLogPath.get.dropRight(1) else auditLogPath.get
-
-        // return validated audit log config for aws
-        auditLogConfig.copy(rawAuditPath = Some(finalAuditLogPath), auditLogFormat = auditLogFormat)
-
+      if (ifFetchFromSystemTable(auditLogConfig)){
+        validateAuditLogConfigsFromSystemTable(auditLogConfig,organization_id)
       } else {
-        val ehConfigOp = auditLogConfig.azureAuditLogEventhubConfig
-        require(ehConfigOp.nonEmpty, "When using Azure, an Eventhub must be configured for audit log retrieval")
-        val ehConfig = ehConfigOp.get
-        val ehPrefix = ehConfig.auditRawEventsPrefix
-
-        val cleanPrefix = if (ehPrefix.endsWith("/")) ehPrefix.dropRight(1) else ehPrefix
-        val rawEventsCheckpoint = ehConfig.auditRawEventsChk.getOrElse(s"${ehPrefix}/rawEventsCheckpoint")
-        // TODO -- Audit log bronze is no longer streaming target -- remove this path
-        val auditLogBronzeChk = ehConfig.auditLogChk.getOrElse(s"${ehPrefix}/auditLogBronzeCheckpoint")
-
-        if (config.debugFlag) {
-          println("DEBUG FROM Init")
-          println(s"cleanPrefix = ${cleanPrefix}")
-          println(s"rawEventsCheck = ${rawEventsCheckpoint}")
-          println(s"auditLogsBronzeChk = ${auditLogBronzeChk}")
-          println(s"ehPrefix = ${ehPrefix}")
-        }
-
-        val ehFinalConfig = ehConfig.copy(
-          auditRawEventsPrefix = cleanPrefix,
-          auditRawEventsChk = Some(rawEventsCheckpoint),
-          auditLogChk = Some(auditLogBronzeChk)
-        )
-
-        // parse the connection string to validate format
-        PipelineFunctions.parseAndValidateEHConnectionString(ehFinalConfig.connectionString, ehFinalConfig.azureClientId.isEmpty)
-        // return validated auditLogConfig for Azure
-        auditLogConfig.copy(azureAuditLogEventhubConfig = Some(ehFinalConfig))
+        validateAuditLogConfigsFromCloud(auditLogConfig)
       }
     }
   }
 
+  def ifFetchFromSystemTable(auditLogConfig: AuditLogConfig): Boolean = {
+    if (auditLogConfig.rawAuditPath.getOrElse("").toLowerCase.equals("system")) {
```

Comment:
you could simply return "auditLogConfig.rawAuditPath.getOrElse("").toLowerCase.equals("system")", no need for the if clause

Reply:
we are checking if the input passed is system or not, and based on its value we will set the source for audit_log_data. We will not ask the users to pass the actual system table name